core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -434,7 +434,7 @@ class SparkContext(config: SparkConf) extends Logging {
     _statusTracker = new SparkStatusTracker(this)
 
     _progressBar =
-      if (_conf.getBoolean("spark.ui.showConsoleProgress", true) && !log.isInfoEnabled) {
+      if (_conf.getBoolean("spark.ui.showConsoleProgress", false) && !log.isInfoEnabled) {
```

Comment:
Now, the default value is false. However, `SparkSubmit` will inject 'true' if `spark.ui.showConsoleProgress` is not defined in our Shell environments.

The original issue was the default behavior of this option. This PR solves that. If you want to stop users from using this option in order to enable `ConsoleProgressBar`, we also are able to change this config completely into some internal one, `spark.internal.ui.showConsoleProgress` in this `SparkContext` class.  In that case, `spark.ui.showConsoleProgress` will converted into `spark.internal.ui.showConsoleProgress` in `SparkSubmit`.